jpa/eclipselink.jpa.test.jse/src/org/eclipse/persistence/jpa/test/mapping/TestUnidirectionalOneToMany.java

```diff
@@ -0,0 +1,137 @@
+/*
+ * Copyright (c) 2018 Oracle and/or its affiliates. All rights reserved.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License v. 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0,
+ * or the Eclipse Distribution License v. 1.0 which is available at
+ * http://www.eclipse.org/org/documents/edl-v10.php.
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR BSD-3-Clause
+ */
+
+// Contributors:
+//     09/12/2018 - Will Dazey
+//       - 391279: Add support for Unidirectional OneToMany mappings with non-nullable values
+package org.eclipse.persistence.jpa.test.mapping;
+
+import java.util.ArrayList;
+
+import javax.persistence.EntityManager;
+import javax.persistence.EntityManagerFactory;
+
+import org.eclipse.persistence.jpa.test.framework.DDLGen;
+import org.eclipse.persistence.jpa.test.framework.Emf;
+import org.eclipse.persistence.jpa.test.framework.EmfRunner;
+import org.eclipse.persistence.jpa.test.framework.Property;
+import org.eclipse.persistence.jpa.test.mapping.model.CommentA;
+import org.eclipse.persistence.jpa.test.mapping.model.CommentB;
+import org.eclipse.persistence.jpa.test.mapping.model.ComplexIdA;
+import org.eclipse.persistence.jpa.test.mapping.model.ComplexIdB;
+import org.eclipse.persistence.jpa.test.mapping.model.PostA;
+import org.eclipse.persistence.jpa.test.mapping.model.PostB;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+@RunWith(EmfRunner.class)
+public class TestUnidirectionalOneToMany {
+    @Emf(createTables = DDLGen.DROP_CREATE, 
+            classes = { CommentA.class, CommentB.class, ComplexIdA.class, ComplexIdB.class, PostA.class, PostB.class }, 
+            properties = { @Property(name = "eclipselink.cache.shared.default", value = "false") })
+    private EntityManagerFactory emf;
+
+    /**
+     * EclipseLink performs Inserts to UnidirectionalOneToMany mappings through
+     * deferred Updates. EclipseLink first INSERTS null FK values in the owning
+     * table, then schedules UPDATES to update to the correct value.
+     * 
+     * However, this does not work if the field is 'nullable=false'. If that
+     * configuration exists, then the driver will throw an exception when
+     * attempting to INSERT a null value
+     * 
+     * @throws Exception
+     */
+    @Test
+    public void testInsertOneToManyUni() {
+
+        EntityManager em = emf.createEntityManager();
+
+        em.getTransaction().begin();
+
+        PostA post = new PostA(new ComplexIdA(9));
+        em.persist(post);
+        em.flush();
```

Comment:
If this persist/flush sequence is removed and one relies on cascading to occur during the next persist, then there is still insert/update occurring - is this expected?

Reply:
Ok, I updated to reflect what it is supposed to be testing. Essentially my tests hit two situations: 
1. 'PostInsert': Post entity is being inserted WITH it's Comment relationships. This causes the data modification events to be created on Post's 'postInsert'. 
2. 'ObjectAdded': Post entity is being updated to ADD it's Comment relationships. This causes the data modification events to be created on Post's 'postUpdate'.